vcd/datasource_vcd_network_test.go

```diff
@@ -0,0 +1,367 @@
+							len(network.OrgVDCNetwork.Configuration.IPScopes.IPScope) > 0 {
+							availableNetworks[networkType] = networkRec{network.OrgVDCNetwork, parent}
+						}
+					} else {
+						availableNetworks[networkType] = networkRec{network.OrgVDCNetwork, parent}
+					}
+				}
+			}
+		}
+	}
+
+	networksSearched = true
+	return nil
+}
+
+func TestAccVcdNetworkDirectDS(t *testing.T) {
+	// This test requires access to the vCD before filling templates
+	// Thus it won't run in the short test
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	err := getAvailableNetworks()
+	if err != nil {
+		fmt.Printf("%s\n", err)
+		t.Skip("error getting available networks")
+		return
+	}
+	if len(availableNetworks) == 0 {
+		t.Skip("No networks found - data source test skipped")
+		return
+	}
+
+	networkType := "vcd_network_direct"
+	data, ok := availableNetworks[networkType]
+	if !ok {
+		t.Skip("no direct network found ")
+		return
+	}
+
+	template := networkTemplates[networkType]
+	var params = StringMap{
+		"Org":             testConfig.VCD.Org,
+		"VDC":             testConfig.VCD.Vdc,
+		"ExternalNetwork": data.parent,
+		"NetworkName":     data.network.Name,
+		"FuncName":        "TestNetworkDirectDS",
+		"Tags":            "network",
+	}
+	configText := templateFill(template, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { preRunChecks(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: configText,
+				Check: resource.ComposeTestCheckFunc(
+
+					resource.TestCheckOutput("network_org", testConfig.VCD.Org),
+					resource.TestCheckOutput("network_vdc", testConfig.VCD.Vdc),
+					resource.TestCheckOutput("network_name", data.network.Name),
+					resource.TestCheckOutput("external_network", data.parent),
+				),
+			},
+		},
+	})
+}
+
+func TestAccVcdNetworkRoutedDS(t *testing.T) {
+	// This test requires access to the vCD before filling templates
+	// Thus it won't run in the short test
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+	err := getAvailableNetworks()
+
+	if err != nil {
+		fmt.Printf("%s\n", err)
+		t.Skip("error getting available networks")
+		return
+	}
+	if len(availableNetworks) == 0 {
+		t.Skip("No networks found - data source test skipped")
+		return
+	}
+
+	networkType := "vcd_network_routed"
+	data, ok := availableNetworks[networkType]
+	if !ok {
+		t.Skip("no routed network found ")
+		return
+	}
+
+	template := networkTemplates[networkType]
+	var params = StringMap{
+		"Org":         testConfig.VCD.Org,
+		"VDC":         testConfig.VCD.Vdc,
+		"EdgeGateway": data.parent,
+		"NetworkName": data.network.Name,
+		"FuncName":    "TestNetworkRoutedDS",
+		"Tags":        "network",
+	}
+	configText := templateFill(template, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+
+	resource.Test(t, resource.TestCase{
```

Comment:
Worth trying out `resource.ParallelTest` to shave of some test time.

Reply:
done

Reply:
done